homeassistant/components/bond/button.py

```diff
@@ -0,0 +1,241 @@
+    ),
+    BondButtonEntityDescription(
+        key=Action.INCREASE_SPEED,
+        name="Increase Speed",
+        icon="mdi:skew-more",
+        mutually_exclusive=Action.SET_SPEED,
+    ),
+    BondButtonEntityDescription(
+        key=Action.DECREASE_SPEED,
+        name="Decrease Speed",
+        icon="mdi:skew-less",
+        mutually_exclusive=Action.SET_SPEED,
+    ),
+    BondButtonEntityDescription(
+        key=Action.TOGGLE_DIRECTION,
+        name="Toggle Direction",
+        icon="mdi:directions-fork",
+        mutually_exclusive=Action.SET_DIRECTION,
+    ),
+    BondButtonEntityDescription(
+        key=Action.INCREASE_TEMPERATURE,
+        name="Increase Temperature",
+        icon="mdi:thermometer-plus",
+        mutually_exclusive=None,
+    ),
+    BondButtonEntityDescription(
+        key=Action.DECREASE_TEMPERATURE,
+        name="Decrease Temperature",
+        icon="mdi:thermometer-minus",
+        mutually_exclusive=None,
+    ),
+    BondButtonEntityDescription(
+        key=Action.INCREASE_FLAME,
+        name="Increase Flame",
+        icon="mdi:fire",
+        mutually_exclusive=Action.SET_FLAME,
+    ),
+    BondButtonEntityDescription(
+        key=Action.DECREASE_FLAME,
+        name="Decrease Flame",
+        icon="mdi:fire-off",
+        mutually_exclusive=Action.SET_FLAME,
+    ),
+    BondButtonEntityDescription(
+        key=Action.TOGGLE_OPEN, name="Toggle Open", mutually_exclusive=Action.OPEN
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Bond button devices."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    hub: BondHub = data[HUB]
+    bpup_subs: BPUPSubscriptions = data[BPUP_SUBS]
+
+    async_add_entities(
+        [
+            BondButtonEntity(hub, device, bpup_subs, description)
+            for device in hub.devices
+            for description in BUTTONS
+            if device.has_action(description.key)
+            and (
+                description.mutually_exclusive is None
+                or not device.has_action(description.mutually_exclusive)
+            )
+        ],
+        True,
+    )
+
+
+class BondButtonEntity(BondEntity, ButtonEntity):
+    """Bond Button Device."""
+
+    def __init__(
+        self,
+        hub: BondHub,
+        device: BondDevice,
+        bpup_subs: BPUPSubscriptions,
+        description: ButtonEntityDescription,
+    ) -> None:
+        """Init Bond button."""
+        super().__init__(
+            hub, device, bpup_subs, description.name, description.key.lower()
+        )
+        self.entity_description = description
+
+    async def async_press(self, **kwargs: Any) -> None:
+        """Press the button."""
+        await self._hub.bond.action(
+            self._device.device_id, Action(self.entity_description.key)
+        )
+
+    def _apply_state(self, state: dict) -> None:
```

Comment:
The parent class has this as required, but buttons are stateless so we do nothing